Assets/_Laz/Scripts/Environment/Puzzles/Connection/NodeConnectionBehaviour.cs

```diff
@@ -0,0 +1,103 @@
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(Collider))]
+    public class NodeConnectionBehaviour : MonoBehaviour
+    {
+        [SerializeField] private ParticleSystem _particleSystem = null;
+        [SerializeField] private Transform _meshTransform = null;
+        
+        private Node _node = null;
+        private LazoBehaviour _lazo = null;
+        
+        public void Initialize(Node node)
+        {
+            ClearParticleSystem();
+            _node = node;
+            _node.OnCanActivateChanged += HandleOnCanActivateChanged;
+            _node.OnNodeCompleted += HandleOnNodeCompleted;
+        }
+
+        public void CleanUp()
+        {
+            ClearParticleSystem();
+            HandleOnNodeCompleted();
+            _node.OnCanActivateChanged -= HandleOnCanActivateChanged;
+            _node.OnNodeCompleted += HandleOnNodeCompleted;
+            _node = null;
+        }
+
+        public void Reset()
+        {
+            HandleOnCanActivateChanged(false);
+        }
+
+        private void ClearParticleSystem()
+        {
+            _particleSystem.Stop();
+            _particleSystem.Clear();
+        }
+        
+        private void ActivateNodeIfNeeded(LazoBehaviour lazo)
+        {
+            if (lazo is {IsLazoing: true} && _node.CanActivate)
```

Comment:
is this the same as lazo.IsLazoing?

Reply:
yeah. Rider just refactored it for me

it used to be 
lazo != null && lazo.IsLazoing


I never seen this structure before.